File: clustermanage-server/src/main/java/cc/bitky/clustermanage/server/message/base/BaseMessage.java
package cc.bitky.clustermanage.server.message.base;

/**
 * 基础信息 Message
 */
public class BaseMessage implements IMessage {

    private int msgId = -1;
    private int groupId = -1;
    private int deviceId = -1;

    protected BaseMessage(int groupId) {
        this.groupId = groupId;
    }

    protected BaseMessage(int groupId, int deviceId) {
        this(groupId);
        this.deviceId = deviceId;
    }

    @Override
    public int getMsgId() {
        return msgId;
    }

    protected void setMsgId(int msgId) {
        this.msgId = msgId;
    }

    @Override
    public int getGroupId() {
        return groupId;
    }

    @Override
    public void setGroupId(int groupId) {
        this.groupId = groupId;
    }

    @Override
    public int getDeviceId() {
        return deviceId;
    }

    @Override
    public void setDeviceId(int deviceId) {
        this.deviceId = deviceId;
    }
}

File: hardware/SingleDeviceTestNET/SingleDeviceTestNET/message/base/BaseMessage.cs
namespace SocketServer.message.@base
{
    public class BaseMessage : IMessage
    {
        private int _msgId = -1;
        private int _groupId = -1;
        private int _deviceId = -1;

        protected BaseMessage(int groupId)
        {
            this._groupId = groupId;
        }

        protected BaseMessage(int groupId, int deviceId) : this(groupId)
        {
            this._deviceId = deviceId;
        }

        public int getMsgId()
        {
            return _msgId;
        }

        protected void setMsgId(int msgId)
        {
            this._msgId = msgId;
        }

        public int getGroupId()
        {
            return _groupId;
        }

        public void setGroupId(int groupId)
        {
            this._groupId = groupId;
        }


        public int getDeviceId()
        {
            return _deviceId;
        }


        public void setDeviceId(int deviceId)
        {
            this._deviceId = deviceId;
        }
    }
}